### PR TITLE
Include BinaryDilate and BinaryErode functions in unit tests

### PR DESCRIPTION
**Pull request by lrm25**
…ests

### CODE REVIEW

File: test/unit_tests/unit_test_image_function.cpp [hunk 1]
```diff
@@ -351,6 +351,68 @@ namespace Function_Template
...
+
+    bool form2_BinaryDilate(BinaryDilateForm2 BinaryDilate)
+    {
+        std::vector <uint8_t> v (100, 255u);
+        v.push_back(0u);
+        PenguinV_Image::Image image = randomImage(v);
+
+        uint32_t roiX, roiY, roiWidth, roiHeight;
+        generateRoi ( image, roiX, roiY, roiWidth, roiHeight );
+        uint8_t result = !verifyImage (image, roiX, roiY, roiWidth, roiHeight, 0u) ? 255u : 0u;
```
Comment (ihhub) [inline test/unit_tests/unit_test_image_function.cpp]: Could you please explain why do we need to do this?

**Comment by ihhub** [comment]
I've changed full ROI functions (through web editor). Please have a look into changes :)

File: test/unit_tests/unit_test_image_function.cpp (hunk 1 above)
Comment (lrm25) [reply test/unit_tests/unit_test_image_function.cpp]: verifyImage will return white if the random image region can be dilated, and if the region is fully black (and can't be dilated), black will be returned.  I was comparing this to the value returned from BinaryDilate to see if the dilation results matched what I predicted.  If it's not right, let me know why, or if there's an easier or better way, let me know.

I'll pull that makefile out and put it in a new PR.

Comment (ihhub) [reply test/unit_tests/unit_test_image_function.cpp]: Okay. I got the idea. I corrected the logic of functions to be more obvious for other people.

**Comment by ihhub** [comment]
@lrm25 , thank you very much for your continuous support for the project! I hope you would continue to do the same :)

**Pull request merged**